File: pulse-go/internal/pulse/move_type_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestMoveType(t *testing.T) {
	t.Run("MoveTypes array should be valid", func(t *testing.T) {
		for moveType := range MoveTypes {
			if MoveTypes[moveType] != moveType {
				t.Errorf("wanted move types at index %d to be %v, got %v", moveType, moveType, MoveTypes[moveType])
			}
		}
	})
}

File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/MoveTypeTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.moveTypes
import kotlin.test.Test
import kotlin.test.assertEquals

class MoveTypeTest {
	@Test
	fun `MoveTypes array should be valid`() {
		for (moveType in moveTypes.indices) {
			assertEquals(moveTypes[moveType], moveType)
		}
	}
}